content/blog/2020/2020-4-015-covid-19-part-3.md

```diff
@@ -0,0 +1,114 @@
+
+Then, connect As Timeseries widget and Line Chart and let's get plotting.
+
+Once in Line Chart, we can select multiple countries to draw them on the same axis or, we could compare multiple plots by clicking on Add plot.
+{{% figure src="/blog_img/2020/2020-04-15-line-chart1.png"%}}
+
+Here, we see how the virus started its path in China, where the number of cases quickly rose but has since then stayed level. Other countries were hit later on. The slopes of the curves tell us how fast the numbers are growing, but more on that later. One thing to notice though: due to fast growth of the virus in some countries, some smaller slopes are almost invisible. We can remedy that by checking the Logarithmic axis box just like this:
+{{% figure src="/blog_img/2020/2020-04-15-log-linechart.png"%}}
+We can now see the first steps as well as the current situation.
+
+### Python Script
+We can see from these plots, that different countries came into contact with the virus at different times. It seems it would be
+much easier to compare the curves, if all of them started their climb at the same time. We can do that, with just a few lines of code. Example snippet here aligns the curves with the moment where the country recorded 100 cases. You can set n to any other (positive) number if you will though.
+
+```
+import numpy as np
+from copy import deepcopy
+
+n = 100
+out_data = deepcopy(in_data)
+out_data.X[out_data.X < n] = 0
+ar = np.argwhere(out_data.X)
+
+cols, shifts = np.unique(ar[:, 1], return_counts=True)
+out_data.X = np.array([np.roll(out_data.X[:, col], shift) for col, shift in zip(cols, shifts)]).T
+out_data.X[out_data.X==0]= np.nan`
+```
+Copy these lines into Python Script widget, and plug in our initial snake of widgets like so:
+{{% figure src="/blog_img/2020/2020-04-15-snake-pyscript.png"%}}
+
+Look at the aligned curves now, they are much easier to compare.
+{{% figure src="/blog_img/2020/2020-04-15-script-log-lc.png"%}}
+
+
+### Absolute growth with derivative
+Let's take a look at how fast the confirmed cases are spreading. Connect As Timeseries with Difference widget, choose Differencing and select countries of interest. Differencing order of 1 means we'll be looking at derivative of first order, which just means daily change in our case. We'll leave shift as is, on 1. 
+
+{{% sidework src="/blog_img/2020/2020-04-15-diff-settings.png"%}}
+
+Again, we'll view the transformed curves in Line Chart. Notice how the difference transform (and every other) adds a new column. That means we can compare our curves in different states, so be sure to always check what is being shown. 
+For starters, try looking at for example China and its transformed version. See how easy it is to spot the daily spikes, that are perhaps out of the ordinary and need to be checked. The most prominent spike here is probably due to change in counting. 
```

Comment:
I would put commas before and after "for example". And remove the one before "that".